app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -37,11 +44,61 @@ public APIClient() {
+                            try {
+                                Log.d("Type", "ConversionError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Conversion Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.HTTP)) {
+                            try {
+                                Log.d("Type", "HTTPError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("HTTP Error", String.valueOf(cause.getResponse().getStatus()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.UNEXPECTED)) {
+                            try {
+                                Log.d("Type", "UnexpectedError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Unexpected Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.NETWORK)) {
+                            try {
+                                Log.d("Type", "NetworkError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Network Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else {
+                            Log.d("Type", "Other Error");
+                            Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            ErrorDialogEvent errEvent = new ErrorDialogEvent("Other Error", String.valueOf(cause.getCause()));
+                            OpenEventApp.postEventOnUIThread(errEvent);
```

Comment:
I believe that switch-case is pretty efficient in terms of enums, which is the case here, nevertheless I feel switch case is more readable and easier to maintain than an if-then-else.